lib/skywalking/segment_ref.lua

```diff
@@ -31,131 +31,78 @@ local _M = {}
+function _M.fromSW8Value(value)
     local ref = _M.new()
 
     local parts = Util.split(value, '-')
-    if #parts ~= 9 then
+    if #parts ~= 8 then
         return nil
     end
 
-    ref.trace_id = Util.formatID(decode_base64(parts[2]))
-    ref.segment_id = Util.formatID(decode_base64(parts[3]))
+    ref.trace_id = decode_base64(parts[2])
+    ref.segment_id = decode_base64(parts[3])
     ref.span_id = tonumber(parts[4])
-    ref.parent_service_instance_id = tonumber(parts[5])
-    ref.entry_service_instance_id = tonumber(parts[6])
-    local peerStr = decode_base64(parts[7])
-    if string.sub(peerStr, 1, 1) == '#' then
-        ref.network_address = string.sub(peerStr, 2)
-    else
-        ref.network_address_id = tonumber(peerStr)
-    end
-    local entryEndpointStr = decode_base64(parts[8])
-    if string.sub(entryEndpointStr, 1, 1) == '#' then
-        ref.entry_endpoint_name = string.sub(entryEndpointStr, 2)
-    else
-        ref.entry_endpoint_id = tonumber(entryEndpointStr)
-    end
-    local parentEndpointStr = decode_base64(parts[9])
-    if string.sub(parentEndpointStr, 1, 1) == '#' then
-        ref.parent_endpoint_name = string.sub(parentEndpointStr, 2)
-    else
-        ref.parent_endpoint_id = tonumber(parentEndpointStr)
-    end
+    ref.parent_service = decode_base64(parts[5])
+    ref.parent_service_instance = decode_base64(parts[6])
+    ref.parent_endpoint = decode_base64(parts[7])
+    ref.address_used_at_client = decode_base64(parts[8])
 
     return ref
 end
 
 -- Return string to represent this ref.
 function _M.serialize(ref)
     local encodedRef = '1'
-    encodedRef = encodedRef .. '-' .. encode_base64(Util.id2String(ref.trace_id))
-    encodedRef = encodedRef .. '-' .. encode_base64(Util.id2String(ref.segment_id))
+    encodedRef = encodedRef .. '-' .. encode_base64(ref.trace_id)
+    encodedRef = encodedRef .. '-' .. encode_base64(ref.segment_id)
     encodedRef = encodedRef .. '-' .. ref.span_id
-    encodedRef = encodedRef .. '-' .. ref.parent_service_instance_id
-    encodedRef = encodedRef .. '-' .. ref.entry_service_instance_id
-
-    local networkAddress
-    if ref.network_address_id ~= 0 then
-        networkAddress = ref.network_address_id .. ''
-    else
-        networkAddress = '#' .. ref.network_address
-    end
-    encodedRef = encodedRef .. '-' .. encode_base64(networkAddress)
-
-    local entryEndpoint
-    if ref.entry_endpoint_id ~= 0 then
-        entryEndpoint = ref.entry_endpoint_id .. ''
-    else
-        entryEndpoint = '#' .. ref.entry_endpoint_name
-    end
-    encodedRef = encodedRef .. '-' .. encode_base64(entryEndpoint)
-
-    local parentEndpoint
-    if ref.parent_endpoint_id ~= 0 then
-        parentEndpoint = ref.parent_endpoint_id .. ''
-    else
-        parentEndpoint = '#' .. ref.parent_endpoint_name
-    end
-    encodedRef = encodedRef .. '-' .. encode_base64(parentEndpoint)
+    encodedRef = encodedRef .. '-' .. encode_base64(ref.parent_service)
```

Comment:
this style is better performance.

```lua
encodedRef = encodedRef .. '-' .. xxx
             .. '-' .. xxx
             .. '-' .. xxx
```

Reply:
@membphis code styles are low priority things, and I think we should start with the structure to see if it fits with OpenResty and Apache APISIX users

Reply:
I'm learning this project, give me more time.